src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Internal.Logger;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Region
+{
+    internal sealed class RegionDiscoveryProvider : IRegionDiscoveryProvider
+    {
+        private const string RegionName = "REGION_NAME";
+        private readonly Uri _ImdsUri;
+        private IDictionary<string, string> Headers;
+        private readonly ICoreLogger _logger;
+        private readonly IHttpManager _httpManager;
+
+        public RegionDiscoveryProvider(IHttpManager httpManager, ICoreLogger logger = null)
+        {
+            _httpManager = httpManager;
+            _ImdsUri = new Uri("http://169.254.169.254/metadata/instance/compute/api-version=2019-06-01");
+            Headers = new Dictionary<string, string>();
+            Headers.Add("Metadata", "true");
+            _logger = logger ?? new NullLogger();
+
+        }
+
+        public async Task<string> getRegionAsync()
+        {
+            if (!Environment.GetEnvironmentVariable(RegionName).IsNullOrEmpty())
```

Comment:
You'll get a null ref here because if the env variable does not exist, Env.GetEnvVariable return null and `null.IsNullOrEmpty()` throw a null ref 

Please use `!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(RegionName))` ... and if the env variable is not found, then this evalues to `!string.IsNullOrEmpty(null)` which evalues to `false`

Reply:
Done